File: iOS/Kalorie/Core/Networking/FireStone/FoodPortionDTO.swift
//
//  FoodPortionDTO.swift
//  Kalorie
//
//  Created by Josef Antoni on 09.09.2026.
//

import Foundation

struct FoodPortionDTO: Codable {

    // MARK: - Properties

    let name: String
    let grams: Double

    // MARK: - Init

    init(portion: FoodPortionDomain) {
        name = portion.name
        grams = portion.grams
    }

    // MARK: - Functions

    func asDomain() -> FoodPortionDomain {
        FoodPortionDomain(name: name, grams: grams)
    }
}

File: Android/app/src/main/java/antoni/kalorie/core/networking/FoodPortionDTO.kt
package antoni.kalorie.core.networking

import antoni.kalorie.core.models.FoodPortionDomain
import kotlinx.serialization.Serializable

@Serializable
data class FoodPortionDTO(
    val name: String,
    val grams: Double,
) {

    // MARK: - Init

    constructor(portion: FoodPortionDomain) : this(name = portion.name, grams = portion.grams)

    // MARK: - Functions

    fun asDomain(): FoodPortionDomain = FoodPortionDomain(name = name, grams = grams)
}
